Client/src/Components/DatasetUpload/DataSection/DatasetDataTable.tsx

```diff
@@ -0,0 +1,162 @@
+import 'react-data-grid/dist/react-data-grid.css'
+
+import { Box, Button, Grid, Typography } from '@material-ui/core'
+import DataGrid, { SelectColumn, TextEditor } from 'react-data-grid'
+import { IContent, IData, IVariable, defaultVariable } from '../../../Models/Datasets/IDatasetModel'
+import React, { useState } from 'react'
+
+import { EditVariableHeader } from './EditVariableHeader'
+
+interface IProps {
+  data: IData,
+  onDataChange: (newData: IData) => void
+}
+
+export const DatasetDataTable = (props: IProps) => {
+  const [editedVariableIndex, setEditedVariableIndex] = useState(-1)
+  const [selectedRows, setSelectedRows] = useState(() => new Set<React.Key>())
```

Comment:
Whats going on here? Ive never seen new Set<React.key> before

Reply:
good question

Reply:
cleaned up, was a copy paste from react-data-grid docs

Reply:
new Set<React.Key>() its a set data structure of type react key, we use it here to specify a set of numbers that represents the rows that are currently selected

Reply:
concerning the **new** https://www.geeksforgeeks.org/javascript-new-keyword/#:~:text=New%20keyword%20in%20JavaScript%20is,new%20empty%20object%20is%20created.